libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -179,40 +179,30 @@ public boolean areVcsCorrelated(List<VerifiableCredential> vcs)
         return true;
     }
 
-    public boolean areGivenNamesAndDobCorrelated(List<VerifiableCredential> vcs)
+    public boolean areNamesAndDobCorrelated(List<VerifiableCredential> vcs)
             throws HttpResponseExceptionWithErrorBody {
         var successfulVcs = getSuccessfulVcs(vcs);
 
-        if (!checkNamesForCorrelation(
-                getNameProperty(
-                        getIdentityClaimsForNameCorrelation(successfulVcs),
-                        NamePart.NamePartType.GIVEN_NAME))) {
-            LOGGER.error(LogHelper.buildErrorMessage(ErrorResponse.FAILED_NAME_CORRELATION));
-            return false;
-        }
+        var areGivenNamesCorrelated =
+                checkNamesForCorrelation(
+                        getNameProperty(
+                                getIdentityClaimsForNameCorrelation(successfulVcs),
+                                NamePart.NamePartType.GIVEN_NAME));
 
-        if (!checkBirthDateCorrelationInCredentials(successfulVcs)) {
-            LOGGER.error(LogHelper.buildErrorMessage(ErrorResponse.FAILED_BIRTHDATE_CORRELATION));
-            return false;
-        }
-        return true;
-    }
+        var isFamilyNameCorrelated =
+                checkNamesForCorrelation(
+                        getFamilyNameForCoiCheck(
+                                getIdentityClaimsForNameCorrelation(successfulVcs)));
 
-    public boolean areFamilyNameAndDobCorrelatedForCoiCheck(List<VerifiableCredential> vcs)
-            throws HttpResponseExceptionWithErrorBody {
-        var successfulVcs = getSuccessfulVcs(vcs);
-
-        if (!checkNamesForCorrelation(
-                getFamilyNameForCoiCheck(getIdentityClaimsForNameCorrelation(successfulVcs)))) {
-            LOGGER.error(LogHelper.buildErrorMessage(ErrorResponse.FAILED_NAME_CORRELATION));
+        // Given names AND family name cannot both be changed
+        if (!areGivenNamesCorrelated && !isFamilyNameCorrelated) {
             return false;
         }
 
-        if (!checkBirthDateCorrelationInCredentials(successfulVcs)) {
-            LOGGER.error(LogHelper.buildErrorMessage(ErrorResponse.FAILED_BIRTHDATE_CORRELATION));
-            return false;
-        }
-        return true;
+        var isBirthDateCorrelated = checkBirthDateCorrelationInCredentials(successfulVcs);
+
+        return (areGivenNamesCorrelated && isBirthDateCorrelated)
+                || (isFamilyNameCorrelated && isBirthDateCorrelated);
```

Comment:
could you expand a bit more on this? Interested in what you mean by the logic not short-circuiting

Reply:
If you write something like:
```
return isBirthDateCorrelated(vcs) && (isGivenNamesCorrelated(vcs) || isFamilyNameCorrelated(vcs);
```
then if `isBirthDateCorrelated` returns false, it will 'short-circuit' the logic and won't execute any of the other methods (since it's guaranteed to return false regardless) - https://en.wikipedia.org/wiki/Short-circuit_evaluation

If you pull them out into variables first, then they'll all be computed in advance.

Reply:
ohh I see, thanks that makes sense :) I saw the approval before and merged but I can update this as part of another PR